### PR TITLE
Child and caregiver element lists in caregiver panel

### PR DESCRIPTION
**Pull request by mikolajmirko**
+ Fonts/styles adjustments
+ SVG fix

### CODE REVIEW

File: lib/widgets/caregiver/child_list_element.dart [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import 'dart:developer';
+
+import 'package:flutter/material.dart';
+
+import 'package:fokus/utils/app_locales.dart';
+import 'package:fokus/utils/theme_config.dart';
+
+import 'package:fokus/widgets/child_wallet.dart';
+
+class ChildListElement extends StatelessWidget {
```
Comment (stasgora) [inline lib/widgets/caregiver/child_list_element.dart:10]: This is the same as the reward card so maybe worth generalizing the widget and parameters names?

File: lib/widgets/caregiver/caregiver_list_element.dart [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import 'package:flutter/material.dart';
+
+import 'package:fokus/utils/app_locales.dart';
+import 'package:fokus/utils/theme_config.dart';
+
+class CaregiverListElement extends StatelessWidget {
+	final String name;
+
+	CaregiverListElement({
```
Comment (stasgora) [inline lib/widgets/caregiver/caregiver_list_element.dart:11]: Maybe just call them cards?

File: lib/widgets/caregiver/child_list_element.dart [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+										Padding(
+											padding: EdgeInsets.only(bottom: 4.0),
+											child: Row(
+												children: <Widget>[
+													ChildPointsChip(quantity: pointCount)
+												],
+											)
+										),
+									],
+								)
```
Comment (stasgora) [inline lib/widgets/caregiver/child_list_element.dart:69]: I may be going to far with pattern matching :smile: but that column arrangement of big title, small description and some item also repeats a couple of times (like in Child plans)

File: lib/widgets/child_wallet.dart [hunk 4]
```diff
@@ -38,6 +39,26 @@ class ChildWallet extends StatelessWidget {
...
+			backgroundColor: lighten(AppColors.diamondColor, 0.48), 
+			avatar: SvgPicture.asset('assets/image/currency/diamond.svg', width: 24, fit: BoxFit.cover),
+			label: Text((quantity ?? 0).toString(), style: TextStyle(color: AppColors.diamondColor, fontWeight: FontWeight.bold)),
+			visualDensity: VisualDensity.compact
+		);
+	}
+
+}
+
 class ChildCustomHeader extends StatelessWidget {
```
Comment (stasgora) [inline lib/widgets/child_wallet.dart:62]: Why is this widget in the wallet file anyway? It belongs with the header stuff.

File: lib/pages/caregiver/panel_page.dart [hunk 5]
```diff
@@ -26,8 +30,39 @@ class _CaregiverPanelPageState extends State<CaregiverPanelPage> {
 							HeaderActionButton.normal(Icons.add, 'page.caregiverSection.panel.header.addCaregiver', () => { log("Dodaj opiekuna") })
 						]),
 						Container(
-								padding: EdgeInsets.all(8.0),
-								child: Text('Profile dzieci', textAlign: TextAlign.left, style: Theme.of(context).textTheme.headline2)
+              child: Container(
```
Comment (stasgora) [inline lib/pages/caregiver/panel_page.dart:33]: What are those two empty containers for?

File: lib/pages/caregiver/panel_page.dart [hunk 6]
```diff
@@ -26,8 +30,39 @@ class _CaregiverPanelPageState extends State<CaregiverPanelPage> {
...
+													top: AppBoxProperties.sectionPadding,
+													right: AppBoxProperties.screenEdgePadding,
+													left: AppBoxProperties.screenEdgePadding
+												),
+												child: Text(
+													'${AppLocales.of(context).translate("page.caregiverSection.panel.content.caregiverProfiles")} ',
+													style: Theme.of(context).textTheme.headline2
+												)
+											),
+											CaregiverListElement(name: "Dawid")
```
[the remark this answers is not in the document]
Comment (mikolajmirko) [reply lib/pages/caregiver/panel_page.dart]: Segment widget: title (simple or advanced) + list of widgets (one or more) + custom message & action if no elements.

File: lib/widgets/child_wallet.dart (hunk 4 above)
Comment (mikolajmirko) [reply lib/widgets/child_wallet.dart]: Wallet file deleted. Child custom header in app_header file now.
Removed additional wallet pop-up for now.

File: lib/widgets/caregiver/child_list_element.dart (hunk 3 above)
Comment (mikolajmirko) [reply lib/widgets/caregiver/child_list_element.dart]: Advanced custom widget ItemCard for 90% of the prototyped lists.
For use in universal Segment section.

Comment (stasgora) [reply lib/widgets/caregiver/child_list_element.dart]: https://www.youtube.com/watch?v=RQdxHi4_Pvc

**Comment by mikolajmirko** [comment]
I think it's good enough for now. Some overflow problems will pop up later probably, but we need to move on rn.

This branch now have UI for all main caregiver screens, some small additional work is needed for plans and rewards (and avatars/icons for whole app). Imo it's perfect for logic implementation tasks (custom card's subtitles based on db data).

File: lib/widgets/caregiver/child_list_element.dart (hunk 3 above)
Comment (mikolajmirko) [reply lib/widgets/caregiver/child_list_element.dart]:
```
   ^
  / \
 /   \
/     \
|  |  |
|  |  |
\  \  /
 \  \/
 /\  \
/  \  \
|  |  |
|  |  |
\     /
 \   /
  \ /
   v   egment
```

File: lib/widgets/segment.dart [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+				)
+			)
+		);
+	}
+
+	@override
+	Widget build(BuildContext context) {
+		return Container(
+			child: Flexible(
+				child: ListView(
```
Comment (stasgora) [inline lib/widgets/segment.dart:95]: Just one more thing, this needs to stay in the page widget, around the Segments - unless we want this :smile: 
image

Comment (stasgora) [reply lib/widgets/segment.dart:95]: And again, why the Container? :stuck_out_tongue:

Comment (mikolajmirko) [reply lib/widgets/segment.dart]: I like containers, fixed:
New widget: AppSegments (on the same level as AppHeader).

**Pull request merged**